src/sql/src/statements/create.rs

```diff
@@ -166,36 +130,32 @@ impl Display for Partitions {
                 "PARTITION ON COLUMNS ({}) (\n{}\n)",
                 format_list_comma!(self.column_list),
                 format_list_indent!(self.exprs),
-            )
-        } else {
-            write!(f, "")
+            )?;
         }
+        Ok(())
     }
 }
 
 impl Display for CreateTable {
     fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
-        let if_not_exists = self.format_if_not_exists();
-        let name = &self.name;
-        let columns = format_list_indent!(self.columns);
-        let constraints = self.format_constraints();
-        let partitions = self.format_partitions();
-        let engine = &self.engine;
-        let options = self.format_options();
-        let maybe_external = if self.engine == FILE_ENGINE {
-            "EXTERNAL "
-        } else {
-            ""
-        };
-        write!(
-            f,
-            r#"CREATE {maybe_external}TABLE {if_not_exists} {name} (
-{columns},
-{constraints}
-)
-{partitions}ENGINE={engine}
-{options}"#
-        )
+        write!(f, "CREATE TABLE ")?;
+        if self.if_not_exists {
+            write!(f, "IF NOT EXISTS ")?;
+        }
+        writeln!(f, "{} (", &self.name)?;
+        writeln!(f, "{},", format_list_indent!(self.columns))?;
+        writeln!(f, "{}", format_table_constraint(&self.constraints))?;
+        writeln!(f, ")")?;
+        if let Some(partitions) = &self.partitions {
+            writeln!(f, "{partitions}")?;
+        }
+        writeln!(f, "ENGINE={}", &self.engine)?;
+        if !self.options.is_empty() {
+            writeln!(f, "WITH (")?;
+            writeln!(f, "{}", format_list_indent!(self.options))?;
```

Comment:
IIUC we can further make `pub options: Vec<SqlOption>,` to `pub options: OptionMap,` as in `CreateExternalTable`.